CMakeLists.txt

```diff
@@ -0,0 +1,134 @@
+# Copyright 2023 Viam Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# TODO: [required] libviam_rust_utils handling
+# TODO: [required] Dockerfile
+# TODO: Runtime paths and make BUILD_SHARED_LIBS work
+# TODO: Check required versions of third party libraries
+# TODO: Compiler minima
+# TODO: Finding external projects vs building in (e.g grpc)
+# TODO: Better clang-format integration (versions, dependencies, only changed, etc.)
+# TODO: Version from git describe?
+# TODO: Export macros and visibilty annotations
+
+
+# This is a tricky decision. If we set this too old, we lose access to
+# modern CMake facilities that we want to be able to use, like file
+# sets. If we set it too new, then the system CMake can't be used to
+# build the product. Here, we make an opinionated choice to require
+# the very latest stable CMake as of this writing, arguing that for
+# most uses of this product, it is expected to be cross-compiling for
+# the target platform on a development system, meaning that we are not
+# constrained by the version of CMake available on target systems.
+cmake_minimum_required(VERSION 3.25 FATAL_ERROR)
+
+
+# Identify the project.
+project(viam-cpp-sdk
+  VERSION 0.0.0
+  DESCRIPTION "Viam Robotics C++ SDK"
+  HOMEPAGE_URL https://github.com/viamrobotics/viam-cpp-sdk
+  LANGUAGES CXX
+)
+
+
+# Enforce the C++ standard, and disable extensions
+set(CMAKE_CXX_STANDARD 14)
```

Comment:
This is a change in behavior from the existing `Makefile` build system which currently sets the C++ standard to C++11. However, per discussion with @stuqdog and @njooma my understanding is that the intended language minimum is C++14, so I've just made that change.